homeassistant/components/mikrotik/hub.py

```diff
@@ -164,15 +163,16 @@ def update_devices(self):
         device_list = {}
         try:
             self.all_devices = self.get_list_from_interface(DHCP)
-            if self.support_wireless or self.support_capsman:
+            support_capsman = bool(self.command(MIKROTIK_SERVICES[IS_CAPSMAN]))
```

Comment:
Do we need to run it on every update?

Reply:
I think we can move this to `self.get_hub_details`. Also `self.support_wireless` can be moved their.

Reply:
sorry for my laziness.
i've moved both to a dedicated setup method

Reply:
I think it is better to move them under `self.get_hub_details` and remove this setup method.

Reply:
done :+1: